test/versioned/kafkajs/kafka.tap.js

```diff
@@ -168,23 +168,27 @@ tap.test('sendBatch records correctly', (t) => {
 
   const { agent, consumer, producer, topic } = t.context
   const message = 'test message'
+  const expectedName = 'produce-tx'
 
   agent.on('transactionFinished', (tx) => {
-    const name = `MessageBroker/Kafka/Topic/Produce/Named/${topic}`
-    const segment = tx.agent.tracer.getSegment()
+    if (tx.name === expectedName) {
```

Comment:
I don't understand the changes in this test. Why do we need to wrap the expectation in a conditional?

Reply:
consumers create transactions. We need to check which transaction is getting asserted so I named the producer transactions so we can do that